hudi-common/src/main/java/org/apache/hudi/common/util/HoodieAvroUtils.java

```diff
@@ -214,6 +218,21 @@ private static GenericRecord rewrite(GenericRecord record, Schema schemaWithFiel
     return newRecord;
   }
 
+  /*
+  This function takes the union of all the fields except hoodie metadata fields
+   */
+  private static List<Field> getAllFieldsToWrite(Schema oldSchema, Schema newSchema) {
```

Comment:
rename : `getNewFieldsToWrite()`?

Reply:
actually here it is a union of old and new fields. That is why I kept this name. If you strongly feel about changing the name, let me know.